packages/flutter/test/material/about_test.dart

```diff
@@ -742,6 +742,151 @@ void main() {
     expect(nestedObserver.licensePageCount, 0);
   });
 
+  testWidgets('Barrier dismissible', (WidgetTester tester) async {
+    final AboutDialogObserver rootObserver = AboutDialogObserver();
+    await tester.pumpWidget(
+      MaterialApp(
+        navigatorObservers: <NavigatorObserver>[rootObserver],
+        home: Material(
+          child: Center(
+            child: Builder(
+              builder: (BuildContext context) {
+                return ElevatedButton(
+                  child: const Text('X'),
+                  onPressed: () => showAboutDialog(
+                    context: context,
+                  ),
+                );
+              },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Open the dialog.
+    await tester.tap(find.byType(ElevatedButton));
+    await tester.pumpAndSettle();
+    expect(rootObserver.dialogCount, 1);
+
+    // Tap on the barrier.
+    await tester.tapAt(const Offset(10.0, 10.0));
+    await tester.pumpAndSettle();
+    expect(rootObserver.dialogCount, 0);
+
+    await tester.pumpWidget(
```

Comment:
Nit: Maybe split this test into two here since they're independent?  And same for other tests like this.